### DISCUSSION TITLE
`supabase.functions.invoke()`: Return data with error

### DISCUSSION THREAD

**Discussion by cohlar** [supabase/supabase#11886] [Feature Requests]
When invoking an edge function with the `supabase` client, we receive either `data` (if status code is `2XX`) or `error`, but never both.
However sometimes it is useful to return meaningful data together with an error status code. For example APIs with limited tokens would return an object with the number of remaining tokens and the token refill rate under status code `429`.

**Comment by cohlar** [comment]
If anyone ever stumbles across this discussion, the way to get the body from the error is `await new Response(error.context.body).json()`.

**Reply by ChuckJonas** [reply]
This save me so much time 🙇‍♂️  This should be added to the docs!
